src/transformStateWithClones.js

```diff
@@ -64,8 +64,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const result = [];
+  let transformed = { ...state };
+
+  for (const transform of transforms) {
+    switch (transform.operation) {
+      case 'addProperties':
+        Object.assign(transformed, transform.properties);
+        result.push({ ...transformed });
```

Comment:
you can write this code only ones outside of the switch

Reply:
done